app/controllers/milkshakes_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class MilkshakesController < ApplicationController
+
+  def index
+    vegetarian = if params[:vegetarian] == "no"; false else true end
+    @menu_items = Milkshake.all
+    @menu_items = @menu_items.where(category: params[:category]) if params[:category].present? && params[:category] != "all"
+    @menu_items = @menu_items.where(vegetarian: vegetarian) if params[:vegetarian].present?
+    @menu_items = @menu_items.where("price <= ?", params[:price]) if params[:price].present?
+    @categories = MilkshakeCategories.options
```

Comment:
How about refactoring this into a module/concern that filters ActiveRecords? Something like this cand be included in any model you want to apply filtering to
```
results = where(nil)

filtering_params.each do |key, value|
  results = results.public_send("by_#{key}", value) if value.present?
end

results
```
And then you can create scopes for each filter like 
`scope :by_category, -> (category) { where(category: category) }`